packages/components/src/icon-button/index.js

```diff
@@ -45,7 +45,7 @@ function IconButton( props, ref ) {
 			// the children are empty and...
 			( ! children || ( isArray( children ) && ! children.length ) ) &&
 			// the tooltip is not explicitly disabled.
-			false !== tooltip
+			tooltip !== false
```

Comment:
Readability improvement.

Reply:
Noting that this will probably conflict with changes currently proposed in #19193 . Regardless if it's an improvement, I might suggest to remove it here, since it's not entirely relevant (or close in proximity) to the intended changes.